workshop/exercise-6/README.md

```diff
@@ -25,80 +29,137 @@ metadata:
-        subset: v2
-  - route:
-    - destination:
-        host: guestbook
-        subset: v1
+    - match:
+        - headers:
+            user-agent:
+              regex: '.*Firefox.*'
+      route:
+        - destination:
+            host: guestbook
+            subset: v2
+    - route:
+        - destination:
+            host: guestbook
+            subset: v1
+```
+
+In Istio `VirtualService` rules, there can be only one rule for each service and therefore when defining multiple [HTTPRoute](https://istio.io/docs/reference/config/istio.networking.v1alpha3.html#HTTPRoute) blocks, the order in which they are defined in the yaml matters.  Hence, the original `VirtualService` rule is modified rather than creating a new rule.  With the modified rule, incoming requests originating from `Firefox` browsers will go to the newer version of guestbook.  All other requests fall-through to the next block, which routes all traffic to the original version of guestbook.  
+
+The newer version of the guestbook service call the Watson Tone Analyzer service created in [Exercise 3](../exercise-3/README.md).  By default Istio blocks calls to services outside the service mesh.  In order for calls to reach the Watson service, create the following `ServiceEntry`:
+
+```console
+  kubectl apply -f serviceentry-tone.yaml
+```
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: bluemix-tone-analyzer
+spec:
+  hosts:
+  - "gateway.watsonplatform.net"
+  ports:
+  - number: 80
+    name: https
+    protocol: http
+  resolution: DNS
+  endpoints:
+   - address: gateway.watsonplatform.net
+     ports:
+       https: 443
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: bluemix-tone-analyzer
+spec:
+  host: "gateway.watsonplatform.net"
+  trafficPolicy:
+    tls:
+      mode: SIMPLE # initiates HTTPS when talking to gateway.watsonplatform.net
 ```
 
-In Istio `VirtualService` rules, there can be only one rule for each service and therefore when defining multiple [HTTPRoute](https://istio.io/docs/reference/config/istio.networking.v1alpha3.html#HTTPRoute) blocks, the order in which they are defined in the yaml matters.  With the modified rule, incoming requests going to hostname `guestbook` that contain a header `istio-test-header` with value `version2` will go to the newer version of guestbook.  All other requests fall-through to the next block, which routes all traffic to the original version of guestbook.  This way the same performance tests and functionality tests can be used to run against version 1 and version 2 just by appending a header.
+The `ServiceEntry` defines addresses and ports services within the mesh are allowed to make requests to.  A `DestinationRule` is also created.  This is required because the guestbook service makes calls over HTTP, but the Watson service only accepts HTTPS connections.  The `DestinationRule` tells Istio to convert the outgoing HTTP request to HTTPS.
```

Comment:
We should change the guestbook service to just call `https://gateway.watsonplatform.net`.
Then we don't need the `DestinationRule` and the `ServiceEntry` is simpler:
```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: bluemix-tone-analyzer
spec:
  hosts:
  - "gateway.watsonplatform.net"
  ports:
  - number: 443
    name: https
    protocol: HTTPS
  resolution: DNS
```